versioned_docs/version-5.x/configuring-links.md

```diff
@@ -0,0 +1,355 @@
+  );
+}
+
+function App() {
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={Home} />
+      <Stack.Screen name="Settings" component={Settings} />
+    </Stack.Navigator>
+  );
+}
+```
+
+Here we have a stack navigator in the root, and inside the `Home` screen of the root stack, we have a tab navigator with various screens. With this structure, let's say we want the path `/users/:id` to go to the `Profile` screen. We can express the nested config like so:
+
+```js
+{
+  Home: {
+    screens: {
+      Profile: 'users/:id',
+    },
+  },
+}
+```
+
+In this config, we specify that the `Profile` screen should be resolved for the `users/:id` pattern and it's nested inside the `Home` screen. Then parsing `users/jane` will result in the following state object:
+
+```js
+const state = {
+  routes: [
+    {
+      name: 'Home',
+      state: {
+        routes: [
+          {
+            name: 'Profile',
+            params: { id: 'jane' },
+          },
+        ],
+      },
+    },
+  ],
+};
+```
+
+Sometimes we want to ensure that a certain screen will always be present for the navigator in the state object. We can achieve it by specifying the `initialRouteName` property for that navigator in the config. For the above example, if we want the `Notifications` screen to be the initial route in `Home` tab navigator, we should specify such config (`Notifications` screen needn't be mentioned in `screens` property):
+
+```js
+{
+  Home: {
+    initialRouteName: 'Notifications',
+    screens: {
+      Profile: 'users/:id',
+      Notifications: 'notify/:user',
+    },
+  },
+};
+```
+
+Then, the path `/users/42` will resolve to the following state object:
+
+```js
+const state = {
+  routes: [
+    {
+      name: 'Home',
+      state: {
+        index: 1,
+        routes: [
+          {
+            name: 'Notifications',
+          },
+          {
+            name: 'Profile',
+            params: { id: 'jane' },
+          },
+        ],
+      },
+    },
+  ],
+};
+```
+
+Notice that we can't pass any params to the `Notifications` screen if it isn't explicitly mentioned in the URL string, so the screen should implement handling lack of these params with e.g. providing default ones.
+
+For some advanced cases, specifying the mapping may not be sufficient. You can implement your custom parser to address these cases using the `getStateFromPath` option:
+
+```js
+const linking = {
+  prefixes: ['https://mychat.com', 'mychat://'],
+  config: {
+    Chat: 'feed/:sort',
+  },
+  getStateFromPath: (path, options) => {
+    // Implement your custom parser and return the navigation state here
+  },
+};
+```
+
+## Example App
+
+### Overview
+
+In the example app, we will use the Expo managed workflow. The guide will focus on creating the deep linking configuration and not on creating the components themselves, but you can always check the full implementation in the [github repo](https://github.com/WoLewicki/DeepLinkingExample).
```

Comment:
we should move this to the `react-navigation` org

Reply:
cc @WoLewicki